dig.go

```diff
@@ -377,7 +377,14 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 			return err
 		}
 	}
-	c.dg.Add(n.paramList.DotNodes(), n.resultList.DotNodes())
+
+	constructor := &dot.Ctor{
+		Name:    n.location.Name,
+		Package: n.location.Package,
+		File:    n.location.File,
+		Line:    n.location.Line,
+	}
+	c.dg.Add(constructor, n.paramList.DotNodes(), n.resultList.DotNodes())
```

Comment:
It looks like introducing the `internal/dot` package made it difficult to to interface between this package and the `dot.Graph` type (since we aren't able to pass in the `node` type to the other package , we have to pre-process these types). Let's consider alternatives here.